### PR TITLE
Add prod warning to Composite Aggregation

### PR DESCRIPTION
**Pull request by stefnestor**
The composite aggregation is considered expensive. Users should perform load testing before deploying it in production.

~[7.14 update](http://github.com/elastic/elasticsearch/pull/74559) exacerbated performance concerns on composite aggregations & [fix in discussion](https://github.com/elastic/elasticsearch/pull/78313) is either reverting or will come in 7.16. In the mean time users need disclaimer that before change this search was expensive & esp. now or w/upgrade may impact cluster performance.~

### Preview
https://elasticsearch_78723.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/search-aggregations-bucket-composite-aggregation.html

### CODE REVIEW

**Review by jrodewig — COMMENTED** [review]
Thanks for raising this @stefnestor. I pushed some commits to simplify the wording. Since I contributed, I'll leave final approval to @nik9000.

I'll also clean up the PR description. IIUC the composite agg is still considered expensive, regardless of any issues or fixes related to #74559.

**Review by nik9000 — APPROVED** [review]
LGTM. It is an expensive agg. Not the worst, but you should totally profile with real data if you are going to use this in any hot path or user facing stuff.
